### DISCUSSION TITLE
Activate/deactivate a egg variable

### DISCUSSION THREAD

**Discussion by HerrSammyDE** [pelican-dev/panel#63] [Ideas]
I would like it if the user could decide whether the variable should be overwritten via the panel or not.

As an admin, you should still be able to specify whether the user can activate/deactivate the variable. Because, for example, you should not be able to deactivate an additional query port.

Here is an example:
![chrome_qRahMJ04wE](https://github.com/pelican-dev/panel/assets/38103916/7a83f851-f44a-454a-8a27-6935fb0105d7)

**Comment by DerLev** [comment]
If I understand that correctly, you want there to be a default set by the admin that the user can decide to use or not?
Because a read only variable is already implemented:

![grafik](https://github.com/pelican-dev/panel/assets/33765680/c7b9461a-0370-43d6-b2a4-743dff8eb4af)

![grafik](https://github.com/pelican-dev/panel/assets/33765680/c306a5f6-b7b3-4d9c-9c39-ad4e37aa1d28)

**Reply by HerrSammyDE** [reply]
No, that's not what I mean. The function should stay, it's good!

The user should be able to decide whether the variable should be overwritten (/ used) by the panel or not, unless it is set to "Read Only"
If you can also set this via the API, it would be really nice!

**Comment by lancepioch** [comment]
Thanks, you can just hard code those values into the egg instead!
